{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Single Cell Sequencing\n",
    "## Creates h5ad files \n",
    "\n",
    "\n",
    "Uses exon data extracted from: zUMIs_output\\expression\\*.dgecounts.rds and gene data from: zUMIs_output\\expression\\*.gene_names.txt<br>\n",
    "Creates h5ad file from anndata objects. Each file represents data from one plate (sample)<br>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !pip install --quiet anndata\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import anndata as ad\n",
    "import pyreadr\n",
    "import config as general_paths\n",
    "import os\n",
    "import warnings\n",
    "warnings.filterwarnings(\"ignore\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "figure_path = general_paths.figures_path\n",
    "data_path = general_paths.data_path\n",
    "output_path = general_paths.output_path\n",
    "local_path = general_paths.local_path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "rds_file_sample1 = os.path.join(local_path, \"sample1/counts_umi/sample1_R.rds\") # the file read and prepared in R\n",
    "h5ad_file_sample1 = os.path.join(local_path, \"sample1/counts_umi/sample1_R.h5ad\")  # the file that will be used for Melisas script\n",
    "\n",
    "rds_file_sample2 = os.path.join(local_path, \"sample2/counts_umi/sample2_R.rds\")  # the file read and prepared in R\n",
    "h5ad_file_sample2 = os.path.join(local_path, \"sample2/counts_umi/sample2_R.h5ad\")  # the file that will be used for Melisas script\n",
    "\n",
    "rds_file_sample3 = os.path.join(local_path, \"sample3/counts_umi/sample3_R.rds\")  # the file read and prepared in R\n",
    "h5ad_file_sample3 = os.path.join(local_path, \"sample3/counts_umi/sample3_R.h5ad\")  # the file that will be used for Melisas script\n",
    "\n",
    "rds_file_sample4 = os.path.join(local_path, \"sample4/counts_umi/sample4_R.rds\")  # the file read and prepared in R\n",
    "h5ad_file_sample4 = os.path.join(local_path, \"sample4/counts_umi/sample4_R.h5ad\")  # the file that will be used for Melisas script\n",
    "\n",
    "rds_file_sample5 = os.path.join(local_path, \"sample5/counts_umi/sample5_R.rds\")  # the file read and prepared in R\n",
    "h5ad_file_sample5 = os.path.join(local_path, \"sample5/counts_umi/sample5_R.h5ad\")  # the file that will be used for Melisas script\n",
    "\n",
    "rds_file_sample6 = os.path.join(local_path, \"sample6/counts_umi/sample6_R.rds\")  # the file read and prepared in R\n",
    "h5ad_file_sample6 = os.path.join(local_path, \"sample6/counts_umi/sample6_R.h5ad\")  # the file that will be used for Melisas script\n",
    "\n",
    "rds_file_sample7 = os.path.join(local_path, \"sample7/counts_umi/sample7_R.rds\")  # the file read and prepared in R\n",
    "h5ad_file_sample7 = os.path.join(local_path, \"sample7/counts_umi/sample7_R.h5ad\")  # the file that will be used for Melisas script\n",
    "\n",
    "# file with gene ids and names from kallisto\n",
    "t2g_file_path = os.path.join(local_path, \"t2g.txt\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Read the RDS file and convert to a DataFrame\n",
    "rds_sample1 = pyreadr.read_r(rds_file_sample1)\n",
    "\n",
    "rds_sample2 = pyreadr.read_r(rds_file_sample2)\n",
    "\n",
    "rds_sample3 = pyreadr.read_r(rds_file_sample3)\n",
    "\n",
    "rds_sample4 = pyreadr.read_r(rds_file_sample4)\n",
    "\n",
    "rds_sample5 = pyreadr.read_r(rds_file_sample5)\n",
    "\n",
    "rds_sample6 = pyreadr.read_r(rds_file_sample6)\n",
    "\n",
    "rds_sample7 = pyreadr.read_r(rds_file_sample7)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# convert to dataframes (they are different sizes)\n",
    "rds_sample1_df = rds_sample1[list(rds_sample1.keys())[0]]\n",
    "\n",
    "rds_sample2_df = rds_sample2[list(rds_sample2.keys())[0]]\n",
    "\n",
    "rds_sample3_df = rds_sample3[list(rds_sample3.keys())[0]]\n",
    "\n",
    "rds_sample4_df = rds_sample4[list(rds_sample4.keys())[0]]\n",
    "\n",
    "rds_sample5_df = rds_sample5[list(rds_sample5.keys())[0]]\n",
    "\n",
    "rds_sample6_df = rds_sample6[list(rds_sample6.keys())[0]]\n",
    "\n",
    "rds_sample7_df = rds_sample7[list(rds_sample7.keys())[0]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AACCATCGGCAACTACCACT</th>\n",
       "      <th>AACCATCGGCAAGTTATCGG</th>\n",
       "      <th>AACCATCGGCCACAATCCAC</th>\n",
       "      <th>AACCATCGGCCACTAACCGG</th>\n",
       "      <th>AACCATCGGCCCGTTCGTAT</th>\n",
       "      <th>AACCATCGGCCGACGGTTGT</th>\n",
       "      <th>AACCATCGGCCGAGTCATTA</th>\n",
       "      <th>AACCATCGGCGAAGCTAGCC</th>\n",
       "      <th>AACCATCGGCGTAACGTCTA</th>\n",
       "      <th>AACCATCGGCTCCAATGCAA</th>\n",
       "      <th>...</th>\n",
       "      <th>TTGTTACTCCCGGTTGGCTA</th>\n",
       "      <th>TTGTTACTCCGAATGAACGT</th>\n",
       "      <th>TTGTTACTCCGAGCGGAACT</th>\n",
       "      <th>TTGTTACTCCTGACAGAGGT</th>\n",
       "      <th>TTGTTACTCCTGCAGGCTAC</th>\n",
       "      <th>TTGTTACTCCTTACCGGACC</th>\n",
       "      <th>TTGTTACTCCTTATGCCGCT</th>\n",
       "      <th>TTGTTACTCCTTATTCGCCT</th>\n",
       "      <th>TTGTTACTCCTTCTCGCCTC</th>\n",
       "      <th>TTGTTACTCCTTGCTAAGGA</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rownames</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSMUSG00000000001.4</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSMUSG00000000028.15</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSMUSG00000000031.16</th>\n",
       "      <td>16.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSMUSG00000000037.17</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSMUSG00000000049.11</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSMUSG00000118124.1</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSMUSG00000118151.1</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSMUSG00000118223.1</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSMUSG00000118284.1</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSMUSG00000118469.1</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>41321 rows Ã— 2688 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                       AACCATCGGCAACTACCACT  AACCATCGGCAAGTTATCGG  \\\n",
       "rownames                                                            \n",
       "ENSMUSG00000000001.4                    0.0                   1.0   \n",
       "ENSMUSG00000000028.15                   0.0                   0.0   \n",
       "ENSMUSG00000000031.16                  16.0                   0.0   \n",
       "ENSMUSG00000000037.17                   0.0                   0.0   \n",
       "ENSMUSG00000000049.11                   0.0                   0.0   \n",
       "...                                     ...                   ...   \n",
       "ENSMUSG00000118124.1                    0.0                   0.0   \n",
       "ENSMUSG00000118151.1                    0.0                   0.0   \n",
       "ENSMUSG00000118223.1                    0.0                   0.0   \n",
       "ENSMUSG00000118284.1                    0.0                   0.0   \n",
       "ENSMUSG00000118469.1                    0.0                   0.0   \n",
       "\n",
       "                       AACCATCGGCCACAATCCAC  AACCATCGGCCACTAACCGG  \\\n",
       "rownames                                                            \n",
       "ENSMUSG00000000001.4                    4.0                   0.0   \n",
       "ENSMUSG00000000028.15                   0.0                   0.0   \n",
       "ENSMUSG00000000031.16                   0.0                   0.0   \n",
       "ENSMUSG00000000037.17                   0.0                   0.0   \n",
       "ENSMUSG00000000049.11                   0.0                   0.0   \n",
       "...                                     ...                   ...   \n",
       "ENSMUSG00000118124.1                    0.0                   0.0   \n",
       "ENSMUSG00000118151.1                    0.0                   0.0   \n",
       "ENSMUSG00000118223.1                    0.0                   0.0   \n",
       "ENSMUSG00000118284.1                    0.0                   0.0   \n",
       "ENSMUSG00000118469.1                    0.0                   0.0   \n",
       "\n",
       "                       AACCATCGGCCCGTTCGTAT  AACCATCGGCCGACGGTTGT  \\\n",
       "rownames                                                            \n",
       "ENSMUSG00000000001.4                    2.0                   4.0   \n",
       "ENSMUSG00000000028.15                   0.0                   0.0   \n",
       "ENSMUSG00000000031.16                   0.0                   0.0   \n",
       "ENSMUSG00000000037.17                   0.0                   0.0   \n",
       "ENSMUSG00000000049.11                   0.0                   0.0   \n",
       "...                                     ...                   ...   \n",
       "ENSMUSG00000118124.1                    0.0                   0.0   \n",
       "ENSMUSG00000118151.1                    0.0                   0.0   \n",
       "ENSMUSG00000118223.1                    0.0                   0.0   \n",
       "ENSMUSG00000118284.1                    0.0                   0.0   \n",
       "ENSMUSG00000118469.1                    0.0                   0.0   \n",
       "\n",
       "                       AACCATCGGCCGAGTCATTA  AACCATCGGCGAAGCTAGCC  \\\n",
       "rownames                                                            \n",
       "ENSMUSG00000000001.4                    2.0                   2.0   \n",
       "ENSMUSG00000000028.15                   0.0                   0.0   \n",
       "ENSMUSG00000000031.16                   2.0                   0.0   \n",
       "ENSMUSG00000000037.17                   0.0                   0.0   \n",
       "ENSMUSG00000000049.11                   0.0                   0.0   \n",
       "...                                     ...                   ...   \n",
       "ENSMUSG00000118124.1                    0.0                   0.0   \n",
       "ENSMUSG00000118151.1                    0.0                   0.0   \n",
       "ENSMUSG00000118223.1                    0.0                   0.0   \n",
       "ENSMUSG00000118284.1                    0.0                   0.0   \n",
       "ENSMUSG00000118469.1                    0.0                   0.0   \n",
       "\n",
       "                       AACCATCGGCGTAACGTCTA  AACCATCGGCTCCAATGCAA  ...  \\\n",
       "rownames                                                           ...   \n",
       "ENSMUSG00000000001.4                    1.0                   0.0  ...   \n",
       "ENSMUSG00000000028.15                   0.0                   0.0  ...   \n",
       "ENSMUSG00000000031.16                   0.0                   0.0  ...   \n",
       "ENSMUSG00000000037.17                   0.0                   0.0  ...   \n",
       "ENSMUSG00000000049.11                   0.0                   0.0  ...   \n",
       "...                                     ...                   ...  ...   \n",
       "ENSMUSG00000118124.1                    0.0                   0.0  ...   \n",
       "ENSMUSG00000118151.1                    0.0                   0.0  ...   \n",
       "ENSMUSG00000118223.1                    0.0                   0.0  ...   \n",
       "ENSMUSG00000118284.1                    0.0                   0.0  ...   \n",
       "ENSMUSG00000118469.1                    0.0                   0.0  ...   \n",
       "\n",
       "                       TTGTTACTCCCGGTTGGCTA  TTGTTACTCCGAATGAACGT  \\\n",
       "rownames                                                            \n",
       "ENSMUSG00000000001.4                    2.0                   6.0   \n",
       "ENSMUSG00000000028.15                   2.0                   0.0   \n",
       "ENSMUSG00000000031.16                   0.0                   1.0   \n",
       "ENSMUSG00000000037.17                   0.0                   0.0   \n",
       "ENSMUSG00000000049.11                   0.0                   0.0   \n",
       "...                                     ...                   ...   \n",
       "ENSMUSG00000118124.1                    0.0                   0.0   \n",
       "ENSMUSG00000118151.1                    0.0                   0.0   \n",
       "ENSMUSG00000118223.1                    0.0                   0.0   \n",
       "ENSMUSG00000118284.1                    0.0                   0.0   \n",
       "ENSMUSG00000118469.1                    0.0                   0.0   \n",
       "\n",
       "                       TTGTTACTCCGAGCGGAACT  TTGTTACTCCTGACAGAGGT  \\\n",
       "rownames                                                            \n",
       "ENSMUSG00000000001.4                    2.0                   2.0   \n",
       "ENSMUSG00000000028.15                   0.0                   0.0   \n",
       "ENSMUSG00000000031.16                   3.0                   2.0   \n",
       "ENSMUSG00000000037.17                   0.0                   0.0   \n",
       "ENSMUSG00000000049.11                   0.0                   1.0   \n",
       "...                                     ...                   ...   \n",
       "ENSMUSG00000118124.1                    0.0                   0.0   \n",
       "ENSMUSG00000118151.1                    0.0                   0.0   \n",
       "ENSMUSG00000118223.1                    0.0                   0.0   \n",
       "ENSMUSG00000118284.1                    0.0                   0.0   \n",
       "ENSMUSG00000118469.1                    0.0                   0.0   \n",
       "\n",
       "                       TTGTTACTCCTGCAGGCTAC  TTGTTACTCCTTACCGGACC  \\\n",
       "rownames                                                            \n",
       "ENSMUSG00000000001.4                    1.0                   3.0   \n",
       "ENSMUSG00000000028.15                   0.0                   0.0   \n",
       "ENSMUSG00000000031.16                   0.0                   0.0   \n",
       "ENSMUSG00000000037.17                   0.0                   0.0   \n",
       "ENSMUSG00000000049.11                   0.0                   0.0   \n",
       "...                                     ...                   ...   \n",
       "ENSMUSG00000118124.1                    0.0                   0.0   \n",
       "ENSMUSG00000118151.1                    0.0                   0.0   \n",
       "ENSMUSG00000118223.1                    0.0                   0.0   \n",
       "ENSMUSG00000118284.1                    0.0                   0.0   \n",
       "ENSMUSG00000118469.1                    0.0                   0.0   \n",
       "\n",
       "                       TTGTTACTCCTTATGCCGCT  TTGTTACTCCTTATTCGCCT  \\\n",
       "rownames                                                            \n",
       "ENSMUSG00000000001.4                    2.0                   4.0   \n",
       "ENSMUSG00000000028.15                   0.0                   0.0   \n",
       "ENSMUSG00000000031.16                   1.0                   0.0   \n",
       "ENSMUSG00000000037.17                   0.0                   0.0   \n",
       "ENSMUSG00000000049.11                   0.0                   0.0   \n",
       "...                                     ...                   ...   \n",
       "ENSMUSG00000118124.1                    0.0                   0.0   \n",
       "ENSMUSG00000118151.1                    0.0                   0.0   \n",
       "ENSMUSG00000118223.1                    0.0                   0.0   \n",
       "ENSMUSG00000118284.1                    0.0                   0.0   \n",
       "ENSMUSG00000118469.1                    0.0                   0.0   \n",
       "\n",
       "                       TTGTTACTCCTTCTCGCCTC  TTGTTACTCCTTGCTAAGGA  \n",
       "rownames                                                           \n",
       "ENSMUSG00000000001.4                    5.0                   4.0  \n",
       "ENSMUSG00000000028.15                   1.0                   0.0  \n",
       "ENSMUSG00000000031.16                   0.0                   0.0  \n",
       "ENSMUSG00000000037.17                   0.0                   0.0  \n",
       "ENSMUSG00000000049.11                   2.0                   0.0  \n",
       "...                                     ...                   ...  \n",
       "ENSMUSG00000118124.1                    0.0                   0.0  \n",
       "ENSMUSG00000118151.1                    0.0                   0.0  \n",
       "ENSMUSG00000118223.1                    0.0                   0.0  \n",
       "ENSMUSG00000118284.1                    0.0                   0.0  \n",
       "ENSMUSG00000118469.1                    0.0                   0.0  \n",
       "\n",
       "[41321 rows x 2688 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# merge whole dataframe, so that all will have the same amount of genes\n",
    "unified_df = pd.concat([rds_sample1_df,rds_sample2_df,rds_sample3_df,rds_sample4_df,rds_sample5_df,rds_sample6_df,rds_sample7_df],axis=1)\n",
    "# replace all NaN values with zero counts\n",
    "unified_df = unified_df.fillna(0.0)\n",
    "unified_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# t2g = pd.read_csv(t2g_file_path, header=None, names=[\"tid\", \"gene_id\", \"gene_name\"], sep=\"\\t\")\n",
    "# t2g.index = t2g.gene_id### FOR THAT WE USE T2G VARIABLE THAT OPENS T2G FILE (FILE WITH LINK BETWEEN GENE CODE AND GENE NAME)\n",
    "\n",
    "# t2g = t2g.loc[~t2g.index.duplicated(keep='first')]\n",
    "# gene_ids_kallisto = t2g['gene_id'].tolist()\n",
    "# gene_ids_scilifelab = list(unified_df.index)\n",
    "# diffelence1 = list(set(gene_ids_kallisto) - set(gene_ids_scilifelab))\n",
    "# diffelence2 = list(set(gene_ids_scilifelab) - set(gene_ids_kallisto))\n",
    "# print(f\"Gene ids that are in kallisto file but not in scilifelab: {len(diffelence1)}\\nGene ids that are in scilifelab but not in kallisto: {len(diffelence2)}\") \n",
    "# # there are 308 gene ids from scilifelab that are not in kallisto ids!\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(384, 384, 384, 384, 384, 384, 384)"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# split back to 3 original dataframes but same numbers of genes now\n",
    "total_barcodes_df1 = rds_sample1_df.shape[1]\n",
    "total_barcodes_df2 = rds_sample2_df.shape[1]\n",
    "total_barcodes_df3 = rds_sample3_df.shape[1]\n",
    "total_barcodes_df4 = rds_sample4_df.shape[1]\n",
    "total_barcodes_df5 = rds_sample5_df.shape[1]\n",
    "total_barcodes_df6 = rds_sample6_df.shape[1]\n",
    "total_barcodes_df7 = rds_sample7_df.shape[1]\n",
    "\n",
    "rds_sample1_df = unified_df.iloc[:,0:total_barcodes_df1]\n",
    "till_col = total_barcodes_df1+total_barcodes_df2\n",
    "rds_sample2_df = unified_df.iloc[:,total_barcodes_df1:till_col]\n",
    "from_col = till_col\n",
    "till_col = from_col+total_barcodes_df3\n",
    "rds_sample3_df = unified_df.iloc[:,from_col:till_col]\n",
    "from_col = till_col\n",
    "till_col = from_col+total_barcodes_df4\n",
    "rds_sample4_df = unified_df.iloc[:,from_col:till_col]\n",
    "from_col = till_col\n",
    "till_col = from_col+total_barcodes_df5\n",
    "rds_sample5_df = unified_df.iloc[:,from_col:till_col]\n",
    "from_col = till_col\n",
    "till_col = from_col+total_barcodes_df6\n",
    "rds_sample6_df = unified_df.iloc[:,from_col:till_col]\n",
    "from_col = till_col\n",
    "till_col = from_col+total_barcodes_df7\n",
    "rds_sample7_df = unified_df.iloc[:,from_col:till_col]\n",
    "\n",
    "\n",
    "rds_sample1_df.shape[1],rds_sample2_df.shape[1],rds_sample3_df.shape[1],rds_sample4_df.shape[1],rds_sample5_df.shape[1],rds_sample6_df.shape[1],rds_sample7_df.shape[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# transpose before creating anndata object, so that gene ids are columns and barcodes are rows\n",
    "transposed_rds_sample1_df = rds_sample1_df.transpose()# Create an AnnData object\n",
    "adata1 = ad.AnnData(\n",
    "    X=transposed_rds_sample1_df.values,  # Your data as numpy array\n",
    "    obs={'barcode': transposed_rds_sample1_df.index},\n",
    "    var={'gene_id': transposed_rds_sample1_df.columns}  \n",
    ")\n",
    "\n",
    "transposed_rds_sample2_df = rds_sample2_df.transpose()# Create an AnnData object\n",
    "adata2 = ad.AnnData(\n",
    "    X=transposed_rds_sample2_df.values,  # Your data as numpy array\n",
    "    obs={'barcode': transposed_rds_sample2_df.index},\n",
    "    var={'gene_id': transposed_rds_sample2_df.columns}  \n",
    ")\n",
    "\n",
    "transposed_rds_sample3_df = rds_sample3_df.transpose()# Create an AnnData object\n",
    "adata3 = ad.AnnData(\n",
    "    X=transposed_rds_sample3_df.values,  # Your data as numpy array\n",
    "    obs={'barcode': transposed_rds_sample3_df.index},\n",
    "    var={'gene_id': transposed_rds_sample3_df.columns}  \n",
    ")\n",
    "\n",
    "transposed_rds_sample4_df = rds_sample4_df.transpose()# Create an AnnData object\n",
    "adata4 = ad.AnnData(\n",
    "    X=transposed_rds_sample4_df.values,  # Your data as numpy array\n",
    "    obs={'barcode': transposed_rds_sample4_df.index},\n",
    "    var={'gene_id': transposed_rds_sample4_df.columns}  \n",
    ")\n",
    "\n",
    "transposed_rds_sample5_df = rds_sample5_df.transpose()# Create an AnnData object\n",
    "adata5 = ad.AnnData(\n",
    "    X=transposed_rds_sample5_df.values,  # Your data as numpy array\n",
    "    obs={'barcode': transposed_rds_sample5_df.index},\n",
    "    var={'gene_id': transposed_rds_sample5_df.columns}  \n",
    ")\n",
    "\n",
    "transposed_rds_sample6_df = rds_sample6_df.transpose()# Create an AnnData object\n",
    "adata6 = ad.AnnData(\n",
    "    X=transposed_rds_sample6_df.values,  # Your data as numpy array\n",
    "    obs={'barcode': transposed_rds_sample6_df.index},\n",
    "    var={'gene_id': transposed_rds_sample6_df.columns}  \n",
    ")\n",
    "\n",
    "transposed_rds_sample7_df = rds_sample7_df.transpose()# Create an AnnData object\n",
    "adata7 = ad.AnnData(\n",
    "    X=transposed_rds_sample7_df.values,  # Your data as numpy array\n",
    "    obs={'barcode': transposed_rds_sample7_df.index},\n",
    "    var={'gene_id': transposed_rds_sample7_df.columns}  \n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# save h5ad files that are necessary for further Melisas pipeline\n",
    "adata1.write(h5ad_file_sample1)\n",
    "adata2.write(h5ad_file_sample2)\n",
    "adata3.write(h5ad_file_sample3)\n",
    "adata4.write(h5ad_file_sample4)\n",
    "adata5.write(h5ad_file_sample5)\n",
    "adata6.write(h5ad_file_sample6)\n",
    "adata7.write(h5ad_file_sample7)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "scs",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.0"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
